scheduler/plugin/plugin_test.go

```diff
@@ -984,6 +984,99 @@ wasm stack trace:
 	}
 }
 
+func TestReserve(t *testing.T) {
+	tests := []struct {
+		name                   string
+		guestURL               string
+		args                   []string
+		globals                map[string]int32
+		pod                    *v1.Pod
+		nodeName               string
+		expectedStatusCode     framework.Code
+		expectedStatusMessage  string
+		expectedUnreserveError string
+	}{
+		{
+			name:               "Success",
+			pod:                test.PodSmall,
+			nodeName:           "good",
+			expectedStatusCode: framework.Success,
+		},
+		{
+			name:                  "Error",
+			pod:                   test.PodSmall,
+			nodeName:              "bad",
+			expectedStatusCode:    framework.Error,
+			expectedStatusMessage: "name is bad",
+		},
+		{
+			name:     "reachable: flag is 0",
+			guestURL: test.URLTestReserveFromGlobal,
+			pod:      test.PodSmall,
+			nodeName: test.NodeSmall.Name,
+			globals:  map[string]int32{"flag": 0},
+		},
+		{
+			name:                   "unreachable: flag is 1",
+			guestURL:               test.URLTestReserveFromGlobal,
+			pod:                    test.PodSmall,
+			nodeName:               test.NodeSmall.Name,
+			globals:                map[string]int32{"flag": 1},
+			expectedUnreserveError: "wasm: unreserve error: wasm error: unreachable\nwasm stack trace:\n\treserve_from_global.$1()",
+		},
+		{
+			name:                  "panic",
+			guestURL:              test.URLErrorPanicOnReserve,
+			pod:                   test.PodSmall,
+			expectedStatusCode:    framework.Error,
+			expectedStatusMessage: "wasm: reserve error: panic!\nwasm error: unreachable\nwasm stack trace:\n\tpanic_on_reserve.$1() i32",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			guestURL := tc.guestURL
+			if guestURL == "" {
+				guestURL = test.URLTestReserve
+			}
+
+			p, err := wasm.NewFromConfig(ctx, "wasm", wasm.PluginConfig{GuestURL: guestURL, Args: tc.args})
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer p.(io.Closer).Close()
+
+			if len(tc.globals) > 0 {
+				pl := wasm.NewTestWasmPlugin(p)
+				pl.SetGlobals(tc.globals)
+			}
+
+			cycleState := framework.NewCycleState()
+			status := p.(framework.ReservePlugin).Reserve(ctx, cycleState, tc.pod, tc.nodeName)
+			if want, have := tc.expectedStatusCode, status.Code(); want != have {
+				t.Fatalf("unexpected status code: want %v, have %v", want, have)
+			}
+			if want, have := tc.expectedStatusMessage, status.Message(); want != have {
+				t.Fatalf("unexpected status message: want %v, have %v", want, have)
+			}
+			if !status.IsSuccess() {
```

Comment:
nit:
could you add the explanation why you return here if status is not success?
(I guess this is because we don't need to check Unreserve anymore, but leaving a comment here is helpful for others)

Reply:
Yes!
459a5a146c842e5c4730b5b1246bc84280499cca